userspace/libscap/linux/scap_fds.c

```diff
@@ -333,8 +333,15 @@ int32_t scap_fd_handle_regular_file(struct scap_proclist *proclist, char *fname,
 	}
 	else if(fdi->type == SCAP_FD_FILE_V2)
 	{
-		scap_fd_flags_file(fdi, procdir);
-		strlcpy(fdi->info.regularinfo.fname, link_name, sizeof(fdi->info.regularinfo.fname));
+		if (0 == strncmp(link_name, "/memfd:", strlen("/memfd:")))
```

Comment:
Question: why is this here and not above, where we do all the `else if` logic?

Reply:
That's a quirk! All the others file descriptors in /proc/pid/fd are links meanwhile the memfd file descriptors are actually files. With that being said, placing that logic above will result in a piece of code that will be never reached. 😄